vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go

```
@@ -128,13 +129,18 @@ func (s podMetricsSource) List(ctx context.Context, namespace string, opts v1.Li
 	close(customResChan)
 
 	podMetrics := <-resChan
-	if podMetrics == nil || s.customMetricsLister == nil {
+	// Early return if we don't have customMetricsLister
+	if s.customMetricsLister == nil {
+		// If we couldn't find it, return an error
+		if podMetrics == nil {
+			return nil, fmt.Errorf("PodMetrics not found (potentially timed out)")
```

the logic makes sense to me, but the error messages are handwavy. Is timeout the only reason these would fail? How will we know what's the root cause

---

Agreed, let's propagate the error returned by the metrics client, as that should be more descriptive.